### PR TITLE
Rename PETScWrappers::PreconditionerBase to PreconditionBase

### PR DESCRIPTION
**Pull request by elauksap**
Just a class rename, for consistency with [`TrilinosWrappers::PreconditionBase`](https://github.com/dealii/dealii/blob/eebfb45674f2950ebcf8fd24bd07405b5cb6dc41/include/deal.II/lac/trilinos_precondition.h#L80).

### CODE REVIEW

**Review by masterleinad — COMMENTED** [review]
This change is not backward-compatible so we at least need a (deprecated) `using` statement that makes the old name an alias for the new name.

File: include/deal.II/lac/petsc_precondition.h [hunk 1]
```diff
@@ -954,6 +954,12 @@ namespace PETScWrappers
      */
     AdditionalData additional_data;
   };
+
+  /**
+   * Alias for backwards-compatibility.
+   * @deprecated Use PETScWrappers::PreconditionBase instead.
+   */
+  using PreconditionerBase DEAL_II_DEPRECATED = PreconditionBase;
```
Comment (elauksap) [inline include/deal.II/lac/petsc_precondition.h:962]: @masterleinad should be done, thank you!

**Review by masterleinad — COMMENTED** [review]
We also need a changelog entry and let's replace `DEAL_II_DEPRECATED` with `DEAL_II_DEPRECATED_EARLY` to adapt to the new policy.

**Review by masterleinad — APPROVED** [review]
Fine with me.

**Review by bangerth — APPROVED** [review]
Thanks for looking out for the little things such as consistency in a code base! We appreciate these patches!

**Pull request merged**